pangeo_forge_recipes/transforms.py

```diff
@@ -457,14 +469,24 @@ class WriteCombinedReference(beam.PTransform, ZarrWriterMixin):
     .parquet as a storage format.
     """
 
-    store_name: str
+    concat_dims: List[str] = field(default_factory=list)
+    identical_dims: List[str] = field(default_factory=list)
+    mzz_kwargs: dict = field(default_factory=dict)
+    precombine_inputs: bool = False
+    store_name: Optional[str] = None
```

Comment:
`store_name` is required so let's not make that Optional if at all possible.

As for `concat_dims` & `identical_dims`, I think those are both required for `CombineReferences` so I think we should not be using a default factory. If either of these kws are not passed, we _do want_

```python
TypeError: __init__() missing 1 required positional argument:
```
to be raised, which the default factory will prevent.